prow/plank/controller_test.go

```diff
@@ -1512,7 +1512,7 @@ func TestSyncPendingJob(t *testing.T) {
 			},
 			ExpectedState:    prowapi.PendingState,
 			ExpectedComplete: false,
-			ExpectedNumPods:  1,
+			ExpectedNumPods:  0,
```

Comment:
Needed because of https://github.com/kubernetes-sigs/controller-runtime/pull/1399, the fakeclient now handles finalizers by itself so updating an object with deletion timestamp set to remove a finalizer will result in deletion